overlay.py

```diff
@@ -187,8 +188,22 @@ def on_connect(self, client, userdata, flags, rc):
         """ Called automatically when the overlay connects successfully to the
             MQTT broker.
 
-            Overlay implementations may override for one-off operations
-            (e.g. drawing self.base_canvas) """
+            Overlay implementations may override for one-off operations."""
+        pass
+
+    def display_base_layer(self):
```

Comment:
To be consistent with other existing methods `draw_base_layer` might be slightly better here (and below for the protected method)

Reply:
@kdav108 I brought up this same issue when I first looked at the Raspicam, so hopefully I'm able to share this knowledge with you too 😄 . This explanation also includes `update_data_layer()` & `_update_data_layer()` and `on_connect()` & `_on_connect()`.

Basically, `overlay.py` is the parent class and has multiple subclasses that are used for the camera. `display_base_layer()` covers code logic and methods that are shared between **all** camera overlays, so this method is defined in `overlay.py`.  Whereas, `_display_base_layer()` is only used to cover functionality that is **specific** to that overlay. It should be noted that as soon as `display_base_layer()` finishes executing, then `_display_base_layer()` should be called straight after. 

For example, `update_data_layer()` in `overlay.py` is used to catch any errors that may occur while the data layer is being updated. This should be the same for all overlays. But for `_update_data_layer()`, this should vary between overlays. `_update_data_layer()` in `overlay_all_stats.py` should update the data layer at the *top strip*, while `_update_data_layer()` in `overlay_new.py` should update the data layer on the corners. (Note that `overlay_new.py` is our newest one, so obviously this overlay is more desirable to the eye and the code is soooooo much cleaner.) 

I know I started explaining `display_base_layer()` and then used `update_data_layer()` as an example, which is probably not a good thing. But the way these internal functions work is the same.

Reply:
It should also be noted that the non-private method should not be overridden by the sub-classes. `super()` could also work here, but iirc Chris brought up the issue that this would mean that a developer would need to know to use `super()` every time a new overlay is made.